### DISCUSSION TITLE
Disabling runtime hook

### DISCUSSION THREAD

**Discussion by arossert** [pyinstaller/pyinstaller#8298] [PyInstaller] [answered]
Is there a way to disable a runtime hook? 
Specifically, I want to disable `PyInstaller/hooks/rthooks/pyi_rth_win32comgenpy.py` since it is creating some temp dirs for historical reasons that I do not want.

I know that this can be done with "regular" hooks but I could not find anything about runtime hooks except that they are running in a sequence before the builtin hooks.
Do I need to delete it manually before compiling the code to achieve that?

**Comment by ZephyrusBlaze** [comment]
You can simply delete or rename the hook file `pyi_rth_win32comgenpy.py` before compiling the code. This will prevent PyInstaller from including it during the build process, effectively disabling its functionality.

Another option could involve identifying the specific code responsible for creating the unwanted temporary files within the hook file. However, this approach may require a thorough understanding of the code and careful examination of documentation. So, deleting or renaming the hook file appears to be a more straightforward solution.

**Reply by arossert** [reply]
@TheStrange-007 Thanks for the quick replay, this is what I asked, the only way to do it is to delete the hook file?

**Reply by ZephyrusBlaze** [reply, marked as answer]
Yes, deleting or renaming the hook file is the most straightforward way to disable its functionality during the PyInstaller build process.

**Comment by arossert** [comment]
Apparently, it is more complex than that, there is the `rthooks.dat` file that also needs to be modified.
